File: pkg/pwrgrep/testdata/fixtures/c/external-input.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) {
    char line[128];
    // ruleid: c-external-input
    char *home = getenv("HOME");
    // ruleid: c-external-input
    char *first = argv[1];
    // ruleid: c-external-input
    fgets(line, sizeof line, stdin);
    // ok: c-external-input
    char *local = "constant";
    return 0;
}

File: pkg/pwrgrep/testdata/fixtures/python/external-input.py
import os
import sys

from flask import request


def f(parser):
    # ruleid: python-external-input
    home = os.environ["HOME"]
    # ruleid: python-external-input
    user = os.getenv("USER")
    # ruleid: python-external-input
    first = sys.argv[1]
    # ruleid: python-external-input
    args = parser.parse_args()
    # ruleid: python-external-input
    line = input("> ")
    # ruleid: python-external-input
    with open("config.json") as f:
        pass
    # ok: python-external-input
    with open("out.txt", "w") as out:
        pass
    # ruleid: python-external-input
    q = request.args.get("q")
    # ok: python-external-input
    local = "constant"
